packages/utils/upgrade/resources/codemods/5.0.0/entity-service-document-service.code.ts

```diff
@@ -0,0 +1,368 @@
+    publicationState: "preview",
+  },
+];
+
+const objectParam_7 = [entityId, ...objectParam_7_1];
+
+strapi.entityService.findOne(uid, ...objectParam_7);
+
+Case: using a variable declared somewhere else with a partial & nested spread arrays & objects
+
+const objectParam_8_1 = {
+  publicationState: "preview",
+};
+
+const objectParam_8 = [
+  entityId,
+  {
+    fields: ["id", "name", "description"],
+    populate: ["author", "comments"],
+    ...objectParam_8_1,
+  },
+];
+
+strapi.entityService.findOne(uid, ...objectParam_8);
+
+
+Case: some sort of mix of all the above
+
+const objectParam_9_1 = {
+  publicationState: "preview",
+};
+
+const objectParam_9 = {
+  fields: ["id", "name", "description"],
+  populate: ["author", "comments"],
+  ...objectParam_9_1,
+};
+
+strapi.entityService.findOne(uid, ...[entityId, [objectParam_9]]);
+
+Case:  even more complex
+
+const objectParam_10_1 = {
+  publicationState: "preview",
+};
+
+const objectParam_10_2 = [uid, ...[12], ...[objectParam_10_1]];
+const objectParam_10 = [...objectParam_10_2];
+
+strapi.entityService.findOne(...[...objectParam_10]);
+
+*/
+
+const transformDeclaration = (path: ASTPath<any>, name: any, j: JSCodeshift) => {
+  const declaration = findClosesDeclaration(path, name, j);
+
+  if (!declaration) {
+    return;
+  }
+
+  transformElement(path, declaration.init, j);
+};
+
+const transformElement = (path: ASTPath<any>, element: any, j: JSCodeshift) => {
+  switch (true) {
+    case j.ObjectExpression.check(element): {
+      transformObjectParam(path, element, j);
+      break;
+    }
+
+    case j.Identifier.check(element): {
+      transformDeclaration(path, element.name, j);
+      break;
+    }
+
+    case j.SpreadElement.check(element): {
+      transformElement(path, element.argument, j);
+      break;
+    }
+
+    case j.ArrayExpression.check(element): {
+      element.elements.forEach((element) => {
+        transformElement(path, element, j);
+      });
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+};
+
+const transformObjectParam = (path: ASTPath<any>, expression: ObjectExpression, j: JSCodeshift) => {
+  expression.properties.forEach((prop) => {
+    switch (true) {
+      case j.ObjectProperty.check(prop): {
+        if (!j.Identifier.check(prop.key) && !j.Literal.check(prop.key)) {
+          return;
+        }
+
+        if (j.Identifier.check(prop.key) && prop.key.name !== 'publicationState') {
+          return;
+        }
+
+        if (j.Literal.check(prop.key) && prop.key.value !== 'publicationState') {
+          return;
+        }
+
+        if (j.Identifier.check(prop.key) && prop.key.name === 'publicationState') {
+          if (!prop.computed && !prop.shorthand) {
+            prop.key.name = 'status';
+          }
+
+          if (prop.shorthand && !prop.computed) {
+            prop.shorthand = false;
+            prop.key = j.identifier('status');
+            prop.value = j.identifier('publicationState');
```

Comment:
Not sure I understand this one? :thinking:

Reply:
renaming shorthad { publicationState } to { status: publicationState }